src/workerd/util/sqlite.h

```diff
@@ -19,6 +20,20 @@ namespace workerd {
 using kj::byte;
 using kj::uint;
 
+class SqliteObserver {
+public:
+  virtual void addRowsRead(uint64_t rowsRead) {}
```

Comment:
I would combine `addRowsRead()` and `addRowsWritten()` into a single `queryExecuted()` call with two parameters. These are always both called together so it avoids a redundant virtual call, and may make it easier to track other metrics (like counting the number of total queries).